charts/ckan/templates/postgresql/secret.yaml

```diff
@@ -0,0 +1,31 @@
+{{- $name := "postgresql" -}}
+{{- $postgresPassword := (.Values.postgresql.ckanDbs.postgresPassword | default (randAlphaNum 42)) }}
+{{- $ckanDatabasePassword := (.Values.postgresql.ckanDbs.ckanPassword | default (randAlphaNum 42) )}}
+{{- $datastorePassword := (.Values.postgresql.ckanDbs.datastorePassword | default (randAlphaNum 42)) }}
+{{- $datapusherPassword := (.Values.postgresql.ckanDbs.datapusherPassword | default (randAlphaNum 42) )}}
+{{- $ckanDatabaseUsername := (.Values.postgresql.ckanDbs.ckanUsername | default "ckan" )}}
+{{- $ckanDatabase := (.Values.postgresql.ckanDbs.ckanDb | default "ckandb" )}}
+{{- $datastoreUsername := (.Values.postgresql.ckanDbs.datastoreUsername | default "datastore" )}}
+{{- $datastoreDatabase := (.Values.postgresql.ckanDbs.datastoreDb | default "datastoredb" )}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ printf "%s-config" (include "ckan.postgresql.fullname" .) | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: {{ $name }}
+  namespace: {{ .Release.Namespace | quote }}
+stringData:
+  postgresPassword: {{ $postgresPassword | quote }}
+  replicationPassword: {{ (.Values.postgresql.ckanDbs.replicationPassword | default (randAlphaNum 42)) | quote }}
+  ckanDatabaseUsername: {{ $ckanDatabaseUsername | quote }}
+  ckanDatabasePassword: {{ $ckanDatabasePassword | quote }}
+  ckanDatabase: {{ $ckanDatabase | quote }}
+  ckanSqlAlchemyUrl: "postgresql://{{ $ckanDatabaseUsername }}:{{ $ckanDatabasePassword }}@{{ printf "%s-%s" (include "ckan.postgresql.fullname" . ) "primary" }}/{{ $ckanDatabase }}"
```

Comment:
I would think these URLs are better placed in the corresponding deployment with [dependent variables](https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/#define-an-environment-dependent-variable-for-a-container)